engine.go

```diff
@@ -220,6 +225,33 @@ func (e *RTCEngine) handleLocalTrackPublished(res *livekit.TrackPublishedRespons
 	e.trackPublishedChan <- res
 }
 
+func (e *RTCEngine) handleDataPacket(msg webrtc.DataChannelMessage) {
+	packet, err := e.readDataPacket(msg)
+	if err != nil {
+		return
+	}
+	switch msg := packet.Value.(type) {
+	case *livekit.DataPacket_Speaker:
+		if e.OnActiveSpeakersChanged != nil {
+			e.OnActiveSpeakersChanged(msg.Speaker.Speakers)
+		}
+	case *livekit.DataPacket_User:
+		if e.OnDataReceived != nil {
+			e.OnDataReceived(msg.User)
+		}
+	}
+}
+
+func (e *RTCEngine) readDataPacket(msg webrtc.DataChannelMessage) (*livekit.DataPacket, error) {
+	dataPacket := &livekit.DataPacket{}
+	if msg.IsString {
```

Comment:
technically speaking, this can be removed. datapackets are always encoded in proto to ensure the lowest overhead